src/CallChain.js

```diff
@@ -0,0 +1,41 @@
+import chalk from "chalk"
+
+class CallChain {
+  constructor(name1, parent) {
+    this.name = name1
+    this.parent = parent
+  }
+
+  static create(name) {
+    return new CallChain(name)
+  }
+
+  add(name) {
+    return new CallChain(name, this)
+  }
+
+  hasParentName(name) {
+    return this.name === name || (this.parent && this.parent.hasParentName(name))
+  }
+
+  hasCyclic() {
+    const parent = this.parent
```

Comment:
What does this do in this method? Does it do something different in terms of performance or anything than using `this.parent` on lines 23 and 24?